packages/test-utils/src/matches.js

```diff
@@ -5,9 +5,13 @@ import {
 } from 'shared/consts'
 import { isConstructor } from 'shared/validators'
 
-export function vmMatchesName(vm, name) {
+function vmMatchesName(vm, name) {
+  const lc = (name = '') => name.toLowerCase()
+  const lowerCaseName = lc(name)
   return (
-    !!name && (vm.name === name || (vm.$options && vm.$options.name === name))
+    !!name &&
+    (lc(vm.name) === lowerCaseName ||
+      (vm.$options && lc(vm.$options.name) === lowerCaseName))
```

Comment:
I still think we should mirror Vue behavior's here. This will match e.g. `<fooBar>` with `foobar`, `<foobar>` with `FooBar`, although it shouldn't.

Reply:
Hm.. the original issue was that it `find` should not care about casing. I personally think it's fine to disregard casing for making a better DX.

If we decide to match Vue's system, I guess we just close this issue?

Reply:
The original issue was that `<bFormRadioGroup>` doesn't match `BFormRadioGroup`, which Vue does.

Reply:
No it's not, it has nothing to do with the tag in the markup. The issue is finding the component by the `name` field. He was doing `find({ name: 'bFormRadioGroup' })` which was working fine. Then the developer of the Bootstrap lib changed the `name` field to `BFormRadioGroup`, and his test broke, which I think isn't great since the actual behavior of the app was still fine. 

I think that the casing of the name field (and the hyphens too) should not be something the developer needs to think about when using `find`, especially when often the `name` tag is in a thid party library and it's not that easy to see what it is without going to the source code to have a look.

Reply:
OK, last comment, as I don't want to become annoying 🙂: IMO `find[All]('name')` should match exactly what Vue does when trying to find component `<name>` in a template - which is equivalent, I think, to calling `Vue.component('name')`. Case at hand, `find('bFormRadioGroup')` should find `BFormRadioGroup`, because Vue resolves `<bFormRadioGroup>` to `BFormRadioGroup`.

Reply:
I think discussion is good 👍 not annoying, I don't take it personally. 

I think I'm missing something obvious 🤔 . Did you mean to type `find({ name: 'bFormRadioGroup' })`? We don't have any support for `find` for custom tags, because there are some edge cases related to functional component we could not solve in a PR a while ago. So using `find(<string literal>)` only works on regular HTML elements. That's why we have the `name` option as well.
______
Are we in agreement that `find({ name: 'bFormRadioGroup' })` should find a component which has the `name` field set to `BFormRadioGroup`? If so, are you saying that I should update the code to only lowercase the first character? Again, I think I missed the point you were making, you definitely know how Vue resolves component names better than I do.

Opinions aside, let's get this to a point that improves the `find` function (if that means closing this PR and issue because changing the current functionality of `find` is correct, that is fine with me, too). My main goal at the moment is to either fix or close open issues, so only ones that are really problems are remaining.

Reply:
Yes, sorry, I meant `find({ name: 'name' })`, indeed.

Reply:
I shared the link to how Vue resolves component names a few comments back, here it is again for convenience: https://github.com/vuejs/vue/blob/b07087deae060244fe5d9651b38dc5f1e79667a9/src/core/util/options.js#L454-L460 . It basically looks for `'name'`, `camelize('name')` and `capitalize(camelize('name'))`, is this order.

Reply:
Ok great, so what do you think is the best thing to do in this case? I'm pretty confused. Do we just lowercase the first character? The current PR lowercases all the characters. I thought Vue would resolve regardless of casing, I could be wrong on that one, but that's why I did `lc` on all the names.

Reply:
Ok, I get it now. I must have missed the fact it does `capitalize` at the end. I agree, let's just match this. Thanks for walking me through this. 👍

Reply:
done!